AP_HAL_ChibiOS: add GPilot P1 Autopilot support (Remastered) (#33067)
ArduPilot/ardupilot: libraries/AP_HAL_ChibiOS/hwdef/GPILOT_P1/README.md

![G-Pilot P1](gpilot_p1.jpg)

*G-Pilot P1 acts as a highly reliable brain for aircraft, thanks to its dual-processor architecture that provides high processing power. While its main processor rapidly solves complex flight algorithms, the accompanying coprocessor takes control in the event of any system failure, ensuring a safe landing. The triple-redundant sensor group located in its chassis helps the vehicle remain stable, unaffected by in-flight vibrations or magnetic interference. Connecting all sensors via a high-speed data bus allows the vehicle to react to instantaneous changes within milliseconds.*

## Features

* **Processor:** *32bit ARM® STM32H753 Cortex®-M7 (400 MHz / 1 MB RAM / 2 MB FLASH)*
* **Coprocessor:** *32-bit STM32F100 failsafe co-processor*
* **Sensors:**
  * *IMU: 2x ICM-20649 and 1x ICM-20602 (Triple-redundant, 2 sets supported by mechanical shock-absorbing foam, maintained at a constant operating temperature via built-in heaters)*
  * *Barometer: 2x MS-5611*
  * *Compass: 1x RM-3100*
* **BlackBox:** *microSD card slot*
* **Mechanical:** *Dimensions: 57mm x 95.8mm x 37.3mm | Weight: ~130g | Operating Temperature: -40°C ~ +85°C*

> ## Connector and Pin Mappings

Reviewer: we are moving to standarized headings enforce in CI soon

```suggestion
## Pinout
```

Connectors are JST GH 1.25mm pitch, except "Molex ClikMate" POWER1&2 connector.

![G-Pilot P1 Connector Map](pinout1.png)
![G-Pilot P1 Connector Pinout](pinout2.jpg)

### UART Mapping

* *SERIAL0(OTG1) : USB (console - MAVlink2)*

> * *SERIAL1(UART2) : Telem1*

Reviewer: must note if DMA capable on SERIAL1 and up
must show default protocol, connector name additionally is fine

* *SERIAL2(UART3) : Telem2*
* *SERIAL3(UART4) : GPS*
* *SERIAL4(UART8) : GPS*
* *SERIAL5(UART7) : USER*
* *SERIAL6(OTG2) : SLCAN*

The Telem1 and Telem2 ports have RTS/CTS pins, the other UARTs do not
have RTS/CTS.

### PWM Output

*The G-Pilot P1 provides a total of 14 PWM output channels for motors, servos, and other actuators. The outputs are structurally divided to maximize flight reliability:*

> * ***Main Outputs (1-8):** Controlled by the dedicated 32-bit STM32F100 failsafe coprocessor (IO). These channels ensure that critical control surfaces or primary motors remain manageable and can execute failsafe procedures even if the main processor encounters an issue.*

Reviewer: must specify grouping and protocol capability....same for following AUX outputs
add warning on each output in group must have same protocol...see other recent readme....
should really be in format consistent with recent readmes/wiki template

* ***Auxiliary Outputs (9-14):** Directly driven by the primary STM32H753 processor (FMU). These channels offer high-speed updates and are ideal for secondary control surfaces, camera gimbals, or additional peripherals.*

### RC Input

> * SBUS: A dedicated 3-pin 2.54mm pitch header is provided specifically for standard SBUS receivers.

Reviewer: needs wording that this input accepts all unidirectional protocols including PPM

> * CRSF / ELRS: For bidirectional protocols like TBS Crossfire, a full UART connection is required. Users can conveniently connect their receivers to the Telem1, Telem2, or Serial5 ports. To enable this, the corresponding SERIALx_PROTOCOL parameter must be configured to 23 (RCIN) via the ground station software.

Reviewer: ```suggestion
* CRSF / ELRS: For bidirectional protocols like TBS Crossfire, a full UART connection is required. Users can conveniently connect their receivers to the Telem1, Telem2, or Serial5 ports. To enable this, the corresponding SERIALx_PROTOCOL parameter must be configured to 23 (RCIN) via the ground station software. SERIALx_OPTIONS may also need configuring, see [RC systems](https://ardupilot.org/plane/docs/common-rc-systems.html)
```

> ### Interfaces

Reviewer: move this section after FEATURES

* 14x PWM Servo Outputs (8x Co-processor, 6x Processor)
* S.Bus Output
* S.Bus Receiver Input
* Spektrum/DSM Input

* 5x Serial Ports (2x Full Flow Control)
* 2x I2C Ports
* 2x DroneCAN Ports
* 1x SPI Port

* 1x Analog Input
* Safety Button and LED
* High Power Buzzer and Processor Status LED
* SWD Port for Firmware Update

> ### Power Interfaces & Battery Monitor

Reviewer: ```suggestion
### Power Monitor
```

*The G-Pilot P1 features a dedicated power port designed to integrate seamlessly with the included GBRICK LV power module. This setup provides stable power to the system while delivering precise analog voltage and current monitoring.*

* Main Power Input (POWER 1)
  * 4.7V ~ 5.3V DC power input
  * Analog voltage and current sensing
  * Maximum 3.3V for analog sensing pins

* Backup Power Input (POWER 2)
  * 4.7V ~ 5.3V DC power input
  * Analog1 voltage and current sensing
  * Maximum 3.3V for analog sensing pins

> 

Reviewer: These values do not match hwdef defines!
Also BATT2 values should be added and a note that either port can be used with supplied power module

Reviewer: did not add note requested

Author: @Hwurzburg Sorry about that, I fixed the analog text typo and added the note regarding the GBRICK LV compatibility with both ports.

> * BATT_VOLT_PIN 14

Reviewer: ```suggestion
Preconfigured defaults:

* BATT_VOLT_PIN 14
```

Author: Added "Preconfig defaults" sub-header.

Author: @Hwurzburg Done, can you resolve this comment?

* BATT_CURR_PIN 15
* BATT_VOLT_MULT 10.1
* BATT_AMP_PERVLT 17.0

*The firmware is pre-configured with the correct battery monitoring multipliers for the GBRICK LV, ensuring accurate battery telemetry out of the box.*

### Compass

> The GPILOT P1 has a RM-3100 built-in compass.

Reviewer: ```suggestion
The GPILOT P1 has a RM-3100 built-in compass. Often this internal compass is disabled due to power interference and a remotely located compass is used.
```

* **BlackBox:** *microSD card slot*
* **Mechanical:** *Dimensions: 57mm x 95.8mm x 37.3mm | Weight: ~130g | Operating Temperature: -40°C ~ +85°C*

## Interfaces

* 14x PWM Servo Outputs (8x Co-processor, 6x Processor)
* S.Bus Output
* S.Bus Receiver Input
* Spektrum/DSM Input

* 5x Serial Ports (2x Full Flow Control)
* 2x I2C Ports
* 2x DroneCAN Ports
* 1x SPI Port

* 1x Analog Input
* Safety Button and LED
* High Power Buzzer and Processor Status LED
* SWD Port for Firmware Update

## Pinout

Connectors are JST GH 1.25mm pitch, except "Molex ClikMate" POWER1&2 connector.

![G-Pilot P1 Connector Map](pinout1.png)
![G-Pilot P1 Connector Pinout](pinout2.png)

## UART Mapping

|Serial#  | PORT    |Protocol        |Notes            |
|---------|---------|----------------|-----------------|
|0        |  OTG1   |  USB           | Mavlink2        |
|1        |  USART2 |  Telem1        | RTS/CTS Enabled |
|2        |  USART3 |  Telem2        | RTS/CTS Enabled |
|3        |  UART4  |  GPS1          | No DMA (RX)     |
|4        |  UART8  |  GPS2          | No DMA (TX)     |

> |5        |  UART7  |  USER          | DMA             |

Reviewer: Protocok column is connector label, not protocol....having connector label is fine, but a protocol column is needed.....Serial1 and 2 should note DMA capable

Author: Fixed, can you check again?

# G-Pilot P1 Flight Controller

![G-Pilot P1](gpilot_p1.jpg)

> *G-Pilot P1 acts as a highly reliable brain for aircraft, thanks to its dual-processor architecture that provides high processing power. While its main processor rapidly solves complex flight algorithms, the accompanying coprocessor takes control in the event of any system failure, ensuring a safe landing. The triple-redundant sensor group located in its chassis helps the vehicle remain stable, unaffected by in-flight vibrations or magnetic interference. Connecting all sensors via a high-speed data bus allows the vehicle to react to instantaneous changes within milliseconds.*

Reviewer 2: I think you might be over-selling the F100's role here.

Author: @peterbarker Smoothed F100's role can you resolve this and check last updates?

  * *Barometer: 2x MS-5611*
  * *Compass: 1x RM-3100*
  * *Built-in Heaters: Automatically maintains IMU temperature at a default of 45°C to eliminate thermal drift, fully customizable via Full Parameter List*
* **Servo PWM Voltage Selector:** *Controls and toggles the PWM logic voltage level between 3.3V and 5V for servo rail outputs.*
* **BlackBox:** *microSD card slot*
* **Mechanical:** *Dimensions: 57mm x 95.8mm x 37.3mm | Weight: ~130g | Operating Temperature: -40°C ~ +85°C*

## Interfaces

* 14x PWM Servo Outputs (8x Co-processor, 6x Processor)
* S.Bus Output
* S.Bus Receiver Input
* Spektrum/DSM Input

* 5x Serial Ports (2x Full Flow Control)
* 2x I2C Ports
* 2x DroneCAN Ports
* 1x SPI Port

* 1x Analog Input
* Safety Button and LED
* High Power Buzzer and Processor Status LED
* SWD Port for Firmware Update

## Pinout

Connectors are JST GH 1.25mm pitch, except "Molex ClikMate" POWER1&2 connector.

![G-Pilot P1 Connector Map](pinout1.png)
![G-Pilot P1 Connector Pinout](pinout2.png)

## UART Mapping

|Serial#  | PORT    |Connector Label |  Protocol      | DMA Capability |
|---------|---------|----------------|----------------|----------------|

> |0        |  OTG1   |  USB           |  MAVlink2      | DMA            |

Reviewer: not a UART -> does not have DMA

Author: Done, changed.

|1        |  USART2 |  Telem1        |  MAVlink2      | DMA            |
|2        |  USART3 |  Telem2        |  MAVLink2      | DMA            |
|3        |  UART4  |  GPS1          |  GPS           | No DMA (RX)    |
|4        |  UART8  |  GPS1          |  GPS           | No DMA (TX)    |
|5        |  UART7  |  USER          |  None          | DMA            |

> |6        |  OTG2   |  SLCAN         |  CAN           | DMA            |

Reviewer: not a UART -> does not have DMA, and protocol is MAVLink2

Author: Done , changed to No DMA and Protocol is now MAVLink2

  * *Barometer: 2x MS-5611*
  * *Compass: 1x RM-3100*
  * *Built-in Heaters: Automatically maintains IMU temperature at a default of 45°C to eliminate thermal drift, fully customizable via Full Parameter List*
* **Servo PWM Voltage Selector:** *Controls and toggles the PWM logic voltage level between 3.3V and 5V for servo rail outputs.*
* **BlackBox:** *microSD card slot*
* **Mechanical:** *Dimensions: 57mm x 95.8mm x 37.3mm | Weight: ~130g | Operating Temperature: -40°C ~ +85°C*

## Pinout

Connectors are JST GH 1.25mm pitch, except "Molex ClikMate" POWER1&2 connector.

![G-Pilot P1 Connector Map](pinout1.png)
![G-Pilot P1 Connector Pinout](pinout2.png)

## UART Mapping

|Serial#  | PORT    |Protocol        |Notes            |
|---------|---------|----------------|-----------------|
|0        |  OTG1   |  USB           | Mavlink2        |
|1        |  USART2 |  Telem1        | RTS/CTS Enabled |
|2        |  USART3 |  Telem2        | RTS/CTS Enabled |
|3        |  UART4  |  GPS1          | No DMA (RX)     |
|4        |  UART8  |  GPS2          | No DMA (TX)     |

> |5        |  UART7  |  USER          |                 |

Reviewer: ```suggestion
|5        |  UART7  |  USER          |   DMA       |
```

|6        |  OTG2   |  SLCAN         |                 |

The Telem1 and Telem2 ports have RTS/CTS pins, the other UARTs do not
have RTS/CTS.

## PWM Output

> The G-Pilot P1 provides a total of 14 PWM output channels divided into Main (IOMCU) and Auxiliary (FMU) groups.

Reviewer: ```suggestion
The G-Pilot P1 provides a total of 14 PWM output channels divided into PWM pins (from IOMCU co-processor) and AUX (from H7 cpu) pins.  There are 5 grouping of the outputs. Within a group, ALL outputs must be configured for the same protocol (PWM type or DShot type). Grouping is shown below:
```

remove the two following ### level sections....

### Main Outputs (IOMCU)

* **Channels 1–8:** Driven by the dedicated 32-bit IOMCU coprocessor.
* **Protocol Support:** Standard PWM (up to 400Hz), OneShot, and DShot.

### Auxiliary Outputs (FMU)

* **Channels 9–14:** Directly driven by the primary STM32H7 processor (FMU).
* **Protocol Support:** High-speed PWM, OneShot, and DShot.

> | Output Group     | Channels               | Native Timer | Pin Assignment        | Supported Protocols   |

Reviewer: get rid of cpu pin column...meaningless to user
Use only PWM and DSHOT in the protocols column...one shot is a PWM protocol

|------------------|------------------------|--------------|-----------------------|-----------------------|
| **Main Group 1** | Channels 1, 2          | TIM2         | PA0, PA1              | PWM / OneShot / DShot |
| **Main Group 2** | Channels 3, 4          | TIM4         | PB8, PB9              | PWM / OneShot / DShot |
| **Main Group 3** | Channels 5, 6, 7, 8    | TIM3         | PA6, PA7, PB0, PB1    | PWM / OneShot / DShot |
| **Aux Group 1**  | Channels 9, 10, 11, 12 | TIM1         | PE14, PE13, PE11, PE9 | PWM / OneShot / DShot |
| **Aux Group 2**  | Channels 13, 14        | TIM4         | PD13, PD14            | PWM / OneShot / DShot |

> > [!WARNING]

Reviewer: you can remove this warning as redundant

> **Timer Group Protocol Constraint:** When switching output protocols (e.g., enabling DShot for ESCs), all channels sharing the same native hardware timer group must be configured to use the same protocol. For example, you cannot mix standard PWM and DShot on channels within Main Group 3 (Channels 5–8) simultaneously.

* **PWM Voltage Selector:** Users can switch the main servo PWM voltage level between **3.3V** and **5V** via the Full Parameter List using the `BRD_PWM_VOLT_SEL` parameter. By default, it is configured for **3.3V** output (`BRD_PWM_VOLT_SEL = 0`). Setting this parameter to `1` shifts the voltage max to 5V to help reduce ESC noise interference. Please note that this option only affects the **8 main outputs** controlled by the IOMCU, not the **6 auxiliary outputs** driven by the FMU.

> ## Interfaces

Reviewer: move to after the FEATURES section....in the future, we may be requiring a fixed sequence of some headers

* 1x Analog Input
* Safety Button and LED
* High Power Buzzer and Processor Status LED
* SWD Port for Firmware Update

## Power Monitor

*The G-Pilot P1 features a dedicated power port designed to integrate seamlessly with the included GBRICK LV power module. This setup provides stable power to the system while delivering precise analog voltage and current monitoring.*

* Main Power Input (POWER 1)
  * 4.7V ~ 5.3V DC power input
  * Analog voltage and current sensing
  * Maximum 3.3V for analog sensing pins

* Backup Power Input (POWER 2)
  * 4.7V ~ 5.3V DC power input
  * Analog1 voltage and current sensing
  * Maximum 3.3V for analog sensing pins

Preconfigured defaults:

* BATT_VOLT_PIN 14
* BATT_CURR_PIN 15
* BATT_VOLT_MULT 12.02
* BATT_AMP_PERVLT 39.877

* BATT2_VOLT_PIN 13
* BATT2_CURR_PIN 4
* BATT2_VOLT_MULT 12.02
* BATT2_AMP_PERVLT 39.877

*The firmware is pre-configured with the correct battery monitoring multipliers for the GBRICK LV, ensuring accurate battery telemetry out of the box.*

## Compass

The GPILOT P1 has a RM-3100 built-in compass. Often this internal compass is disabled due to power interference and a remotely located compass is used.

## RC Input

> * **RC Input:** A dedicated 3-pin 2.54mm pitch header is provided for receiver connections. Although commonly labeled or used for SBUS, this input automatically detects and accepts all unidirectional RC protocols, including **PPM, SBUS, DSM, and IBUS**.

Reviewer: ```suggestion
* **RC Input:** A dedicated 3-pin 2.54mm pitch header is provided for receiver connections. Although commonly labeled or used for SBUS, this input automatically detects and accepts all ArduPilot unidirectional RC protocols.
```

* CRSF / ELRS: For bidirectional protocols like TBS Crossfire, a full UART connection is required. Users can conveniently connect their receivers to the Telem1, Telem2, or Serial5 ports. To enable this, the corresponding SERIALx_PROTOCOL parameter must be configured to 23 (RCIN) via the ground station software. SERIALx_OPTIONS may also need configuring, see [RC systems](https://ardupilot.org/plane/docs/common-rc-systems.html)

> ## Loading Firmware

Reviewer: needs a line telling where (firmware seerver) and under what folder names the firmware can be found

Author: Added Firmware server destination below the Firmware header.
